components/script/dom/validitystate.rs

```diff
@@ -84,11 +111,33 @@ impl ValidityStateMethods for ValidityState {
 
     // https://html.spec.whatwg.org/multipage/#dom-validitystate-toolong
     fn TooLong(&self) -> bool {
+        if self.element.has_attribute(&local_name!("maxlength")) {
+            let Max = self.element.get_string_attribute(&local_name!("maxlength")).to_string();
+            let v = self.element.get_string_attribute(&local_name!("value")).to_string();
+            let n = self.element.get_string_attribute(&local_name!("name"));
+            let l = v.len().to_i32().unwrap();
+            if l > Max.parse::<i32>().unwrap() {
+                println!("{} is too long for {}, maxlength is {} characters.", v, n, Max);
+                return true;
+            }
```

Comment:
HTMLInputElement already contains the maxlength information in a field, so this check will be more straightforward in HTMLInputElement::validate.